File: hash/skip_list.py

class SkipList:  # LeetCode Q.1206.
    def __init__(self):
        self.numbers_table = dict()

    def search(self, target: int):
        try:
            self.numbers_table[target]

        except KeyError:
            return False

        return True

    def add(self, number: int):
        if number not in self.numbers_table.keys():
            self.numbers_table.update({number: 0})
        self.numbers_table[number] += 1

    def erase(self, number: int):
        try:
            self.numbers_table[number]

        except KeyError:
            return False

        self.numbers_table[number] -= 1
        if self.numbers_table[number] == 0:
            del self.numbers_table[number]
        return True

File: hash/skip_list/skip_list.cpp
#include <unordered_map>
using namespace std;

class Skiplist
{ // LeetCode Q.1206.
private:
    unordered_map<int, int> numsCounts;

public:
    Skiplist() {}

    bool search(int target)
    {
        if (numsCounts.find(target) == numsCounts.end())
            return false;

        if (numsCounts[target] == 0)
            return false;

        return true;
    }

    void add(int num)
    {
        numsCounts[num]++;
    }

    bool erase(int num)
    {
        numsCounts[num]--;
        if (numsCounts[num] == -1) // Num isn't in skiplist.
        {
            numsCounts[num]++; // Back to 0.
            return false;
        }
        return true;
    }
};